Automattic/jetpack
# Blogroll: show errors and better loading state management

This PR adds a spinner when the appender is loading subscriptions. The appender button is shown only after this request finishes.
To avoid showing two spinners this one is hidden when the recommendations are being loaded.
The PR also shows in the frontend possible errors from the requests we use.

<img width="716" alt="Screenshot 2023-10-05 at 16 22 43" src="https://github.com/Automattic/jetpack/assets/7000684/0125f309-4491-4932-85f6-aed90642c016">

Fixes #
## Proposed changes:

*

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* In the editor create a new post and add the blogroll block
* When loading the recommendations there should be only one spinner visible
* Save the post and reload the page. When the subscriptions are loaded you should see a small spinner button before the button appender is shown.
* Provoke the requests we use to cause an error, or return error from the hooks and verify that the error messages are being shown in the block
